src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -769,4 +770,36 @@ private static ReferenceSequenceFile createReference(final File fastaFile) {
         }
         return splits;
     }
+
+    /**
+     * Generates a list of {@link GenomeLoc} instances given the appropriate {@link GenomeLocParser} factory
+     * and a collection of {@link Locatable} instances.
+     *
+     * <p>
+     *     The order in the result list is will correspond to the traversal order in the input collection.
+     * </p>
+     * @param locatables input locatable collection.
+     * @throws IllegalArgumentException if {@code locatable} is {@code null} or contains any {@code null}.
+     * @return never {@code null}. The result is an unmodifiable list.
+     */
+    public static List<GenomeLoc> genomeLocsFromLocatables(final GenomeLocParser parser, final Collection<? extends Locatable> locatables) {
+        if (parser == null) {
+            throw new IllegalArgumentException("the input genome-loc parser cannot be null");
+        }
+        if (locatables == null) {
+            throw new IllegalArgumentException("the input locatable collection cannot be null");
+        };
+        final int size = locatables.size();
+        if (size == 0)
+            return Collections.emptyList();
+        final List<GenomeLoc> result = new ArrayList<>(locatables.size());
+        locatables.forEach(l -> {
+            if (l == null) {
```

Comment:
can you extract this null check outside. Without it, the whole code turns into a nice 1-liner. And the null check can be done using `locatables.stream().anyMatch(l -> l == null)` which is also a nice 1liner